Rust/Rust.sublime-syntax

```diff
@@ -453,39 +499,47 @@ contexts:
     - include: block
 
   type:
-    - match: (?:{{support_type}}|{{identifier}})(?=<)
-      captures:
-        1: support.type.rust
+    # A low-level type. Typically you want type-any-identifier for the full
+    # type grammar.
+    - match: '{{identifier}}(?=<)'
       push: generic-angles
-    - match: \b(Self|i8|i16|i32|i64|i128|isize|u8|u16|u32|u64|u128|usize|f32|f64|bool|char|str)\b
+    - match: \b(Self|{{int_suffixes}}|{{float_suffixes}}|bool|char|str)\b
       scope: storage.type.rust
+    # stdsimd types
+    - match: |-
+        (?x)
+        \b(?:
+          [ium]8x(?:2|4|8|16|32) | [iu]8x64   |
+          [ium]16x(?:2|4|8|16)   | [iu]16x32  | # f16xN types might come later
+          [iumf]32x(?:2|4|8)     | [iuf]32x16 |
+          [iumf]64x(?:2|4)       | [iuf]64x8  |
+          m1x(?:64|32|16|8)                   | # 512-bit masks
+          __m(?:64|128|256)[di]? # __m512 should come later
+        )\b
+      scope: storage.type.rust
+    - match: '\bdyn\b(?!\s*::)(?=\s*(?:\(|{{lifetime}}|{{identifier}}))'
+      scope: storage.type.trait.rust
 
   generic-angles:
     - meta_scope: meta.generic.rust
-    - include: comments
-    - match: '->'
-      scope: punctuation.separator.generic.rust
-      push: generic-angles-contents
+    # -> to cover https://doc.rust-lang.org/std/boxed/trait.FnBox.html
+    - include: support-type
     - match: '>'
       scope: punctuation.definition.generic.end.rust
       pop: true
     - match: '<'
       scope: punctuation.definition.generic.begin.rust
       push: generic-angles-contents
     # Alert the user of a broken generic construct
-    - match: '\S'
+    - match: '(?=\S)'
       scope: invalid.illegal.rust
       pop: true
 
   generic-angles-contents:
     - include: comments
-    - match: \bextern\b
-      scope: keyword.other.rust
-      push:
-        - include: strings
-        - match: '(?=\S)'
-          pop: true
-    - match: '(?=>|[^ \t\n\$=<_+''(),&:\[\]*[:alnum:]])'
+    - include: attribute
+    - include: type-slice-or-array
```

Comment:
Do we propably need to add `- include: strings` here?

See issue #2323

```rust
impl State<"init"> {
//   ^^^^^^^^^^^^^ meta.impl.rust meta.generic.rust
//                ^ meta.impl.rust - meta.generic - meta.block
//                 ^ meta.impl.rust meta.block.rust
//   ^^^^^ entity.name.impl.rust
//        ^ punctuation.definition.generic.begin.rust
//         ^^^^^^ string.quoted.double.rust
//               ^ punctuation.definition.generic.end.rust
//                 ^ punctuation.section.block.begin.rust
    pub fn accumulate(self) -> State<"accumulate"> {

        State { total: self.total }
    }
}
```

Reply:
I've generally been shying away from supporting unstable/unfinished language features like const generics.  There is a tracking issue at https://github.com/rust-lang/rust-enhanced/issues/333 for things like this.  I'm not entirely opposed to supporting them as long as it is relatively maintainable, but for the purpose of this "sync" PR, I'd prefer to avoid adding enhancements like this.

Also note, there are quite a few known bugs tracked in https://github.com/rust-lang/rust-enhanced/issues?q=is%3Aissue+is%3Aopen+label%3AA-syntax.  I don't expect the syntax to be perfect at this stage, though any notes on anything missing are much appreciated!